templates/components/form/fields_macros.html.twig

```diff
@@ -305,6 +400,100 @@
+   {% endif %}
+
+   {% if options.disabled %}
+      {% set options = options|merge({specific_tags: {'disabled': 'disabled'}}) %}
+   {% endif %}
+
+   {% set field %}
+      {{ Dropdown__showYesNo(name, value, -1, {
+         'display': false,
+         'rand': rand,
+         'width': '100%',
+         'class': 'form-select',
+      }|merge(options)) }}
+   {% endset %}
+
+   {{ _self.field(name, field, label, options|merge({'id': 'dropdown_' ~ name ~ '_' ~ options.rand})) }}
+{% endmacro %}
+
+{% macro dropdownItemTypes(name, value, label = '', options = {}) %}
+   {% set options = {'rand': random()}|merge(options) %}
+   {% if options.fields_template.isMandatoryField(name) %}
+      {% set options = {'required': true}|merge(options) %}
+   {% endif %}
+
+   {% if options.disabled %}
+      {% set options = options|merge({specific_tags: {'disabled': 'disabled'}}) %}
+   {% endif %}
+
+   {% set types = options['types']|default([]) %}
+
+   {% set field %}
+      {{ Dropdown__showItemTypes(name, types, {
+         'display': false,
+         'rand': rand,
+         'width': '100%',
+         'class': 'form-select',
+         'value': value
+      }|merge(options)) }}
+   {% endset %}
+
+   {{ _self.field(name, field, label, options|merge({'id': 'dropdown_' ~ name ~ '_' ~ options.rand})) }}
+{% endmacro %}
+
+{% macro dropdownIcons(name, value, label = '', options = {}) %}
+   {% set options = {'rand': random()}|merge(options) %}
+   {% if options.fields_template.isMandatoryField(name) %}
+      {% set options = {'required': true}|merge(options) %}
+   {% endif %}
+
+   {% if options.disabled %}
+      {% set options = options|merge({specific_tags: {'disabled': 'disabled'}}) %}
+   {% endif %}
+
+   {% set field %}
+      {{ Dropdown__dropdownIcons(name, value, constant('GLPI_ROOT')~'/pics/icones', {
+         'display': false,
+         'rand': rand,
+         'width': '100%',
+         'class': 'form-select',
+      }|merge(options)) }}
+   {% endset %}
+
+   {{ _self.field(name, field, label, options|merge({'id': 'dropdown_' ~ name ~ '_' ~ options.rand})) }}
+{% endmacro %}
+
+{% macro dropdownHoursField(name, value, label = '', options = {}) %}
+   {% set options = {'rand': random()}|merge(options) %}
+   {% if options.fields_template.isMandatoryField(name) %}
+      {% set options = {'required': true}|merge(options) %}
+   {% endif %}
+
+   {% if options.disabled %}
+      {% set options = options|merge({specific_tags: {'disabled': 'disabled'}}) %}
+   {% endif %}
+
+   {% set types = options['types']|default([]) %}
+
+   {% set field %}
+      {{ Dropdown__showHours(name, types, {
+         'display': false,
+         'rand': rand,
+         'width': '100%',
+         'class': 'form-select',
+         'value': value
+      }|merge(options)) }}
+   {% endset %}
+
+   {{ _self.field(name, field, label, options|merge({'id': 'dropdown_' ~ name ~ '_' ~ options.rand})) }}
+{% endmacro %}
+
```

Comment:
regarding the whole part of dropdown fields, could you move them into a separate macro template (included in the main one) ?

Reply:
May be possible if fields were defined as blocks, but I don't know how to get macro inheritance to work properly. You have to import macros in the template that is going to use them.